### DISCUSSION TITLE
Automatic adjustment of the camera

### DISCUSSION THREAD

**Discussion by stdgn** [godotengine/godot-proposals#4393] [Editor]
It would be nice to have the camera automatically adjust to the position we are looking at on the 3D screen.

**Comment by Calinou** [comment]
Are you referring to something like https://github.com/godotengine/godot-proposals/issues/910? If not, could you detail further and possibly create a mockup? It's not clear what you're proposing exactly here.

**Reply by stdgn** [reply]
When we press the Ctrl + Alt + Numpad 0 keys in Blender, the camera looks at that location. I think there should be something like that in Godot too. we can quickly switch to the view we want.

For example, when the camera is selected, the camera can be adjusted by pressing a button in the red circle or with a shortcut.
![cam](https://user-images.githubusercontent.com/39684712/163650605-4dc1e4be-bca9-4425-856e-f15c003b9fc4.png)

**Reply by Calinou** [reply]
This is already implemented :slightly_smiling_face: 

Select a 3D node and press <kbd>Ctrl + Alt + M</kbd> (or <kbd>Ctrl + Alt + F</kbd> to only change the rotation). Alternatively, you can use the **Perspective** menu and choose **Align Transform With View** (or **Align Rotation With View**).

**Reply by aXu-AP** [reply]
> Select a 3D node and press Ctrl + Alt + M (or Ctrl + Alt + F to only change the rotation). Alternatively, you can use the **Perspective** menu and choose **Align Transform With View** (or **Align Rotation With View**).

Would this be more intuitive in the Transform menu, where Snap object to floor is found as well? I wasn't aware that the Perpective menu had options to modify nodes in addition to perspective (altough I see the logic here too).

**Reply by Calinou** [reply]
I agree this should be moved to the Transform menu, especially since the option isn't specific to the current 3D viewport. You can split the 3D editor into several viewports, but your selection remains the same no matter the viewport you're currently using.

It might also be worth adding an **Align Position With View** option which only sets the position, not the rotation.
